### ISSUE TITLE
docs: showcase higher-order fairing, and what a pinned patch keeps under smoothing

### ISSUE THREAD

**Issue by KhoiDOO** [KhoiDOO/conquer3d#66]
`smooth()` gained a `locked` mask and `fair()` is new, neither of which the Mesh gallery
shows.

## Scope

**New figure — higher-order fairing.** Reproduce Botsch et al., *Polygon Mesh Processing*,
Figure 4.8: a swept elbow whose straight ends are held fixed while the bend between them is
faired at orders 1, 2 and 3 — membrane ($\Delta x = 0$), thin plate ($\Delta^2 x = 0$) and
minimum variation ($\Delta^3 x = 0$).

**Updated figure — Laplacian smoothing.** Add a fifth panel running the same 100 iterations
with a patch pinned, so the mask's effect is visible beside the unpinned result.

## Constraints

- The tube is generated, not loaded, with ring spacing matched to the circumferential
  spacing so the triangles come out near-equilateral. A sloppy triangulation would put
  artifacts of the discretisation into the result rather than the behaviour of the flow.
- Each order must run to its steady state, not to a fixed iteration budget. The flow is
  stiff and each order needs roughly ten times the steps of the one below, so a fixed budget
  would show convergence speed rather than the surfaces themselves.
- The figure must use uniform weights, because cotangent does not converge for $k \ge 2$
  (#63). Publishing a fixed-iteration cotangent run would present an oscillating
  intermediate state as the thin-plate and minimum-variation surfaces.

## Verification

`docs/generate.py --strict`, plus the measured convergence residual for each order.

**Comment by KhoiDOO** [comment]
## API usage

Fairing a region while holding the rest of the surface fixed:

```python
import torch
from conquer3d.data_structure import TriangleMesh

# locked is True where a vertex is held fixed. Locking the whole complement of
# the region gives many pinned rings, which is what imposes C^(k-1) at its
# border -- a thin hand-pinned ring would give position continuity only.
locked = ~bend_mask

for k, iterations in ((1, 50_000), (2, 400_000), (3, 4_000_000)):
    mesh = TriangleMesh(verts.clone(), faces.clone())
    mesh.fair(
        k=k,                              # 1 membrane, 2 thin plate, 3 min variation
        iterations=iterations,
        damping=0.5 * 2 ** (1 - k),       # stability is guaranteed up to 2**(1-k)
        mode=0,                           # uniform; cotangent does not converge (#63)
        locked=locked.contiguous(),
    )
```

Smoothing everything except a pinned patch:

```python
patch = (verts - centre).norm(dim=-1) < 0.45
mesh.smooth(iterations=100, damping=0.5, mode=1, locked=patch.contiguous())
```

## Advantages

- **Steady states, not a fixed budget.** Each order is run until per-chunk movement falls
  below tolerance, so the panels show the membrane, thin-plate and minimum-variation
  surfaces rather than three snapshots of how fast each flow happens to move.
- **The mesh is generated, not loaded.** Ring spacing matches circumferential spacing, so
  the triangles are near-equilateral and the figure shows the flow rather than artifacts of
  a scanned triangulation.
- **`locked` as the complement, not a ring.** Locking everything outside the region gives
  many pinned rings, which is what actually imposes C^(k-1) at the border. The docstring's
  caveat about un-dilated masks applies to thin masks, not this.
- The smoothing panel's claim is measured, not asserted: mean |H| in the patch is 1.93 free
  against 6.68 pinned.

## Disadvantages and trade-offs

- **The gallery demonstrates a non-default configuration.** `mode=0` is used because
  cotangent does not converge for k >= 2 (#63), while the docstring calls cotangent
  "strongly preferred". This stays wrong-looking until that bug is fixed.
- **Order 3 is asymptotic, not settled.** At 4,000,000 iterations movement is 2.3e-4 and
  halving every 500k steps. Displacement would land near 0.1035 against the published
  0.1026 — under 1%, but not zero.
- **The figure is slow to build**: ~84 s, dominated by order 3's 4M iterations.
- Uniform weights cannot tell an uneven triangulation from a curved surface, which is
  acceptable here only because the tube is deliberately uniform.

## Measurements

Tube: 4,416 vertices, 8,736 faces, 1,440 free in the bend.

| k | surface | iterations | final per-chunk movement | displacement |
|---|---|---|---|---|
| 1 | membrane, Δx = 0 | 50,000 | 3.7e-9 | 0.417 |
| 2 | thin plate, Δ²x = 0 | 400,000 | 3.6e-7 | 0.143 |
| 3 | min variation, Δ³x = 0 | 4,000,000 | 2.3e-4 | 0.103 |

Smoothing, StanfordBunny, 34,834 vertices, 914 pinned at r = 0.45:

| state | mean \|H\| in patch |
|---|---|
| original | 7.0 |
| 100 iterations, free | 1.93 |
| 100 iterations, pinned | 6.68 |

Docs build: 1173/1173, all links resolve, figure build 84 s.

## Technical insights

- **Displacement falls with order**: 0.417, 0.143, 0.103. The membrane minimises area and so
  pulls the bend hard toward a minimal surface; higher orders minimise curvature and its
  variation, and stay progressively closer to the input. This is what makes the three panels
  visually distinct and matches Figure 4.8.
- **Stiffness is inherent, not a tuning problem.** Each order needs roughly 10x the steps of
  the one below, because the slowest mode decays by a factor set by the square of the
  smallest eigenvalue.
- **Picking a patch by "most camera-facing vertex" is a trap.** On the bunny it lands on an
  ear tip, where smoothing degenerates a thin feature rather than erasing surface detail —
  mean |H| of 70 rather than 2. A height penalty puts it on the haunch. Rendering the mask
  was the only thing that caught it.

**Comment by KhoiDOO** [comment]
Implemented and covered by PR #67.

Four commits: the smoothing panel for `locked`, the fairing showcase, the frozen v0.8.5
archive, and the v0.8.6 site rebuild. Verified at 1173/1173 with all links and snippet
imports resolving.

Left open deliberately: the figure uses `mode=0`, so the gallery demonstrates a
non-default, non-recommended configuration until #63 is fixed. That is tracked there, not
here.
